src/automator/tasks/overview.js

```diff
@@ -1,66 +1,91 @@
 angular.module('hbpCollaboratoryAutomator')
-.run(function createOverview(
-  $log, $q, $http, bbpConfig, hbpFileStore, hbpErrorService,
-  hbpCollaboratoryAutomator, hbpCollaboratoryNavStore
-) {
-  hbpCollaboratoryAutomator.registerHandler('overview', overview);
+  .run(function createOverview(
+    $log, $q, $http, bbpConfig, hbpFileStore, hbpErrorService,
+    hbpCollaboratoryAutomator, hbpCollaboratoryNavStore,
+    hbpCollaboratoryAppStore
+  ) {
+    hbpCollaboratoryAutomator.registerHandler('overview', overview);
 
-  /**
-   * Set the content of the overview page using
-   * the content of a file in storage.
-   *
-   * The collab is indicated either by an id in `descriptor.collab` or a
-   * collab object in `context.collab`.
-   *
-   * @memberof hbpCollaboratory.hbpCollaboratoryAutomator.Tasks
-   * @param {object} descriptor the task configuration
-   * @param {object} [descriptor.collab] id of the collab
-   * @param {string} descriptor.entity either a label that can be found in
-   *                 ``context.entities`` or a FileEntity UUID
-   * @param {object} context the current task context
-   * @param {object} [context.collab] the collab in which entities will be copied
-   * @param {object} [context.entities] a list of entities to lookup in for
-   *                   descriptor.entiry value
-   * @return {object} created entities where keys are the same as provided in
-   *                  config.storage
-   */
-  function overview(descriptor, context) {
-    $log.debug("Fill overview page with content from entity");
-    var fetch = {
-      rootNav: hbpCollaboratoryNavStore.getRoot(
-        descriptor.collab || context.collab.id),
-      source: fetchSourceContent(descriptor, context)
-    };
-    return $q.all(fetch)
-    .then(function(results) {
-      var overview = results.rootNav.children[0];
-      return $http.post(bbpConfig.get('api.richtext.v0') + '/richtext/', {
-        ctx: overview.context,
-        raw: results.source
-      }).then(function() {
-        return overview;
-      });
-    });
-  }
+    /**
+     * Set the content of the overview page.
+     * If an 'entity' is specified, it will use the content of that storage file
+     * If an 'app' name is specified, it will use that app for the overview page
+     *
+     * The collab is indicated either by an id in `descriptor.collab` or a
+     * collab object in `context.collab`.
+     *
+     * @memberof hbpCollaboratory.hbpCollaboratoryAutomator.Tasks
+     * @param {object} descriptor the task configuration
+     * @param {object} [descriptor.collab] id of the collab
+     * @param {string} descriptor.entity either a label that can be found in
```

Comment:
You can now use the [] syntax which means the attribute is optional. [descriptor.entity] and [descriptor.app]